src/base/bittorrent/torrentinfo.h

```diff
@@ -57,7 +57,7 @@ namespace BitTorrent
         typedef boost::shared_ptr<const libtorrent::torrent_info> NativeConstPtr;
         typedef boost::shared_ptr<libtorrent::torrent_info> NativePtr;
 
-        explicit TorrentInfo(NativeConstPtr nativeInfo = NativeConstPtr());
+        explicit TorrentInfo(const NativeConstPtr &nativeInfo = NativeConstPtr());
```

Comment:
I think this should be reverted, `boost::shared_ptr` isn't shouldn't be used like this.

Reply:
:+1:
https://stackoverflow.com/questions/17368644/passing-smart-pointers-by-reference
>It's ok to pass a smart pointer by reference, except if it's to a constructor. In a constructor, it's possible to store a reference to the original object, which violates the contract of the smart pointers.

Reply:
Reverted.